src/chart/bar/BarView.js

```diff
@@ -158,16 +174,24 @@ export default echarts.extendChartView({
                 );
             })
             .update(function (newIndex, oldIndex) {
-                var el = oldData.getItemGraphicEl(oldIndex);
+                var itemModel = data.getItemModel(newIndex);
+                var layout = getLayout[coord.type](data, newIndex, itemModel);
+
+                if (drawBackground) {
+                    var bgEl = oldBgEls[oldIndex];
+                    bgEl.useStyle(backgroundModel.getBarItemStyle());
+                    bgEls.push(bgEl);
```

Comment:
Should better be bgEls[newIndex] = bgEl in case that the callback of add and update are called in a different order from the param `newIndex`.